run.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+if [ "${POSTGRES_ENV_POSTGRES_PASSWORD}" == "**Random**" ]; then
+        unset POSTGRES_ENV_POSTGRES_PASSWORD
+fi
+
+POSTGRES_HOST=${POSTGRES_PORT_5432_TCP_ADDR:-${POSTGRES_HOST}}
+POSTGRES_HOST=${POSTGRES_PORT_1_5432_TCP_ADDR:-${POSTGRES_HOST}}
+POSTGRES_PORT=${POSTGRES_PORT_5432_TCP_PORT:-${POSTGRES_PORT}}
+POSTGRES_PORT=${POSTGRES_PORT_1_3306_TCP_PORT:-${POSTGRES_PORT}}
+POSTGRES_USER=${POSTGRES_USER:-${POSTGRES_ENV_POSTGRES_USER}}
+POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-${POSTGRES_ENV_POSTGRES_PASSWORD}}
+
+[ -z "${POSTGRES_HOST}" ] && { echo "=> POSTGRES_HOST cannot be empty" && exit 1; }
+[ -z "${POSTGRES_PORT}" ] && { echo "=> POSTGRES_PORT cannot be empty" && exit 1; }
+[ -z "${POSTGRES_USER}" ] && { echo "=> POSTGRES_USER cannot be empty" && exit 1; }
+[ -z "${POSTGRES_PASSWORD}" ] && { echo "=> POSTGRES_PASSWORD cannot be empty" && exit 1; }
+[ -z "${POSTGRES_DB}" ] && { echo "=> POSTGRES_DB cannot be empty" && exit 1; }
+
+export PGPASSWORD="${POSTGRES_PASSWORD}"
+
+BACKUP_CMD="pg_dump -h ${POSTGRES_HOST} -p ${POSTGRES_PORT} -U ${POSTGRES_USER} -f /backup/\${BACKUP_NAME} ${POSTGRES_DB}"
+
+echo "=> Creating backup script"
+rm -f /backup.sh
+cat <<EOF >> /backup.sh
+#!/bin/bash
+MAX_BACKUPS=${MAX_BACKUPS}
+SLACK=${SLACK_WEBHOOK}
+BACKUP_NAME=dump_\$(date +%d-%m-%Y_%H_%M_%S).sql
+export PGPASSWORD="${POSTGRES_PASSWORD}"
+echo "=> Backup started: \${BACKUP_NAME}"
+if ${BACKUP_CMD} ;then
+    echo "   Backup succeeded"
+    if [ -n "\${SLACK}" ]; then
+      curl -X POST --data-urlencode "payload={\"username\": \"Backup BOT\", \"text\": \"*HA MAINTENANCE* - database backup succeded: file=*\${BACKUP_NAME}*\",}" \${SLACK}
+    fi
+else
+    echo "   Backup failed"
+    rm -rf /backup/\${BACKUP_NAME}
```

Comment:
if we remove it, we are completely blind. We don't log the error, and the file is gone. We have no way to see what happened. Let's `mv` the file and prefix it with `failed_`.

Reply:
and if we rename it, then in the `while` loop below we'll need to filter these `failed_` out.

Reply:
Excellent idea 👍

Reply:
If there is -f used, in case of failure PG_DUMP does not create file or anything. I can use > instead

Reply:
hm, ok, but in case of error - the error message finds its way to `postgres_backup.log`, right? Together with a timestamp / filename?